### PR TITLE
[HttpClient] Unit test to validate span creation in case of redirect

### PR DESCRIPTION
**Pull request by vishweshbankwar**
Fixes #.

## Changes

Follow up https://github.com/open-telemetry/opentelemetry-dotnet/pull/3072#issuecomment-1258451318

In case of .NET Core new span for each redirect is created as per the [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md#http-redirects-examples). Adding the unit test from original [PR](https://github.com/open-telemetry/opentelemetry-dotnet/pull/3072) created by @denisivan0v to validate that. 

Will do a separate follow up for retries.

For significant contributions please make sure you have completed the following items:

* [ ] Appropriate `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

File: test/OpenTelemetry.Instrumentation.Http.Tests/HttpClientTests.Basic.cs [hunk 1]
```diff
@@ -317,6 +326,28 @@ public async Task HttpClientInstrumentationBacksOffIfAlreadyInstrumented()
...
+            using (Sdk.CreateTracerProviderBuilder()
+                       .AddHttpClientInstrumentation()
+                       .AddProcessor(processor.Object)
+                       .Build())
+            {
+                using var c = new HttpClient();
+                await c.GetAsync($"{this.url}redirect");
+            }
+
+            Assert.Equal(7, processor.Invocations.Count); // SetParentProvider/OnStart/OnEnd/OnStart/OnEnd/OnShutdown/Dispose called.
```
Comment (cijothomas) [inline test/OpenTelemetry.Instrumentation.Http.Tests/HttpClientTests.Basic.cs:342]: Would be nice to get this replaced with InMemoryExporter..

**Comment by cijothomas** [comment]
The test is passing, so we were already doing this for redirects and the pending PR is for the retry one only?

**Review by cijothomas — APPROVED** [review]

**Comment by vishweshbankwar** [comment]
> The test is passing, so we were already doing this for redirects and the pending PR is for the retry one only?

That's right. The [check](https://github.com/open-telemetry/opentelemetry-dotnet/blob/86ee906dcd02c562e85814ca07f2a1a15a83dc98/src/OpenTelemetry.Instrumentation.Http/Implementation/HttpHandlerDiagnosticListener.cs#L119) has no impact on redirect as `Httpclient` [clears](https://github.com/dotnet/runtime/blob/1f688ed75133664aa20c838fa451e2ecf9f1f424/src/libraries/System.Net.Http/src/System/Net/Http/DiagnosticsHandler.cs#L102) the headers for redirects anyways.

File: test/OpenTelemetry.Instrumentation.Http.Tests/HttpClientTests.Basic.cs (hunk 1 above)
Comment (vishweshbankwar) [reply test/OpenTelemetry.Instrumentation.Http.Tests/HttpClientTests.Basic.cs:342]: Can do a follow up as this needs to be changed in other tests as well.

**Pull request merged**